### DISCUSSION TITLE
MOOSE Conda Packages Now Compatible with Xcode > 12.4

### DISCUSSION THREAD

**Discussion by cticenhour** [idaholab/moose#19289] [News]
Conda package updates for `moose-mpich`, `moose-petsc`, and `moose-libmesh` were merged today in #19230, and enable full MacOS Big Sur compatibility for previously unsupported Xcode and Command Line Tools versions 12.5, 13, and 13.1. This update also enables support for MacOS Monterey. The Clang compiler has also been updated to 12.0.1.

**Note that this update only applies to Intel Macs. Official M1 Apple Silicon support is on the way! Track the MOOSE issue #18954 if interested in new developments.**

To download the new packages, please activate your MOOSE conda environment and perform the following command:
```
conda update --all
```
where the appropriate versions and build in the update output should be:
```
Package                  Version            Build
=====================================================
moose-libmesh            2021.10.27         build_1
moose-mpich              3.3.2              build_8
moose-petsc              3.15.1             build_3
```

NOTE that if you do not see `build_1` with your libmesh package update, check your python version. If you are using 3.8 or 3.9, you will need to update instead by doing the following:

```
conda install moose-libmesh moose-tools python=3.7
```

due to a current lack of PyQt availability in conda for Python 3.8 and 3.9.

If any issues with the new packages are experienced, please open a new Discussions post!

**Comment by cticenhour** [comment]
@tophmatthews

**Reply by lynnmunday** [reply]
@cticenhour would this update allow us to use xcode as a gui for lldb?  I've never tried xcode but I would like to find a gui for lldb.

**Reply by cticenhour** [reply]
This update ensures compatibility with the new versions of Xcode, which should mean that the built-in LLDB is compatible. As I use LLDB on the command line, I haven't yet tried to do this. 

@loganharbour @lindsayad @friedmud have any of you used the visual LLDB within Xcode on a MOOSE-based project? Or can ping someone who might have?

**Reply by lynnmunday** [reply]
@dewenyushu used it a few years ago.  I don't know if she has tried again since this update.

**Reply by dewenyushu** [reply]
I tried using Xcode for debugging about two years ago before switching to conda. It worked fine with the 'external build system'.  After switching to conda, I could not run it with the same setting. I recall that the issue was no being able to find some python script(s), but I do not remember the exact name.

Now I am mainly using the command line. Not sure if it would work again with the update. But here is my notes for setting up the debugger in Xcode (two years ago), if you maybe interested 
[Debug with Xcode.pdf](https://github.com/idaholab/moose/files/7770658/Debug.with.Xcode.pdf)

**Reply by friedmud** [reply]
No - I've never used the visual debugging capability in XCode.  Command-line forever ;-)

**Comment by lindsayad** [comment]
I’ve only ever used the command line
